ibis/expr/api.py

```diff
@@ -1544,6 +1546,24 @@ def difference(table: ir.Table, *rest: ir.Table, distinct: bool = True):
     return table.difference(*rest, distinct=distinct) if rest else table
 
 
+def watermark(time_col: str, allowed_delay: ir.IntervalScalar) -> Watermark:
+    """Return a watermark object.
+
+    Parameters
+    ----------
+    time_col:
+        The timestamp column that will be used to generate watermarks in event time processing.
+    allowed_delay:
+        Length of time that events are allowed to be late.
```

Comment:
Docstring formatting nit: no need for `:` here

```suggestion
    time_col
        The timestamp column that will be used to generate watermarks in event time processing.
    allowed_delay
        Length of time that events are allowed to be late.
```